lib/v2/screens/explore_screens/vote/interactor/viewmodels/vote_state.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:equatable/equatable.dart';
+import 'package:seeds/v2/datasource/remote/model/proposals_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+
+/// --- STATE
+class VoteState extends Equatable {
+  final PageState pageState;
+  final Map<String, Map> proposalTypes;
+  final Map<String, String>? currentType;
+  final List<ProposalModel>? proposals;
+  final String? errorMessage;
+
+  const VoteState({
+    required this.pageState,
+    required this.proposalTypes,
+    this.currentType,
+    this.proposals,
+    this.errorMessage,
+  });
+
+  @override
+  List<Object?> get props => [
+        pageState,
+        currentType,
+        proposals,
+        errorMessage,
+      ];
+
+  VoteState copyWith({
+    PageState? pageState,
+    List<ProposalModel>? proposals,
+    Map<String, String>? currentType,
+    String? errorMessage,
+  }) {
+    return VoteState(
+      pageState: pageState ?? this.pageState,
+      proposalTypes: proposalTypes,
+      currentType: currentType ?? this.currentType,
+      proposals: proposals,
+      errorMessage: errorMessage ?? this.errorMessage,
+    );
+  }
+
+  factory VoteState.initial() {
+    return const VoteState(
+      pageState: PageState.initial,
+      proposalTypes: {
```

Comment:
can these proposal types be a class?

Reply:
yeah, I also want but I need to understand what does these map filtres means.